magma/compile.py

```diff
@@ -88,9 +90,13 @@ def __compile_to_coreir(main, file_name, opts):
 def compile(basename, main, output='verilog', **kwargs):
     opts = kwargs.copy()
 
+    mantle_imported = "mantle" in sys.modules
+
     # Rather than having separate logic for 'coreir-verilog' mode, we defer to
     # 'coreir' mode with the 'output_verilog' option set to True.
-    if output == 'coreir-verilog':
+    if output == 'coreir-verilog' or \
```

Comment:
It would be good to have some logging here that we are doing this implicit transformation of `output`. Also maybe we can move this logic outside of this `if` and above do:

```python
if output == "verilog" and m.mantle_target == "coreir" and mantle_imported:
    output = "coreir-verilog"
```